codegen/header_propagate.go

```diff
@@ -75,27 +76,58 @@ func (hp *HeaderPropagator) Propagate(
 			return err
 		}
 
-		// TODO support more primitive types
-		// currently we only support string, and struct that can
-		// convert string, for example: typedef string UUID
-		// future support:
-		// e.g: -H "version: 3.0" -> int64(3)
-
-		assignedVal := "key"
-		if gotype != "string" {
-			assignedVal = gotype + "(key)"
-		}
+		arrs := typeSwitch(key, gotype, field)
+		hp.append(arrs...)
 
-		if !field.Required {
-			hp.appendf(`in.%s = &%s`, key, assignedVal)
-		} else {
-			hp.appendf(`in.%s = %s`, key, assignedVal)
-		}
 		hp.append("}")
 	}
 	return nil
 }
 
+// typeSwitch supports primary type parsing for headers
+func typeSwitch(key, gotype string, field *compile.FieldSpec) []string {
+	var (
+		ret       = []string{}
+		typeParse string
+		typeCast  string
+		assignVal = "key"
+	)
+	switch gotype {
+	case "bool":
+		typeParse = "strconv.ParseBool(key)"
+		assignVal = "v"
+	case "int32":
+		typeParse = "strconv.ParseInt(key, 10, 32)"
+		assignVal = "val"
+		typeCast = "val := int32(v)\n"
+	case "int64":
+		typeParse = "strconv.ParseInt(key, 10, 64)"
+		assignVal = "v"
+	case "float64":
+		typeParse = "strconv.ParseFloat(key, 64)"
+		assignVal = "v"
+	case "string":
+	default:
+		typeCast = "val := " + gotype + "(key)\n"
```

Comment:
This can potentially cause runtime panics, e.g. `val := int16("42")`

Reply:
good catch